std/math.d

```diff
@@ -5884,14 +5885,285 @@ T nextafter(T)(const T x, const T y) @safe pure nothrow @nogc
 real fdim(real x, real y) @safe pure nothrow @nogc { return (x > y) ? x - y : +0.0; }
 
 /****************************************
- * Returns the larger of x and y.
+ * Returns the largest (most positive) element of `nums`.
+ *
+ * Elements are compared after implicit conversion to a common floating-point
+ * type. If `nums` is empty or any element `isNaN`, a `nan` value will be returned.
  */
-real fmax(real x, real y) @safe pure nothrow @nogc { return x > y ? x : y; }
+auto fmax(T ...)(T nums)
+    if (allSatisfy!(ApplyRight!(isImplicitlyConvertible, real), T))
+{
+    pragma(inline, true);
+
+    alias F = AutoFImumType!T;
+    static if (T.length <= 4)
+        return fimumImpl!(">", T.length, F)(nums);
+    else
+        return fimumImpl!(">", F)(nums);
+}
+
+/// ditto
+auto fmax(F, T ...)(T nums)
+    if (isFloatingPoint!F && allSatisfy!(ApplyRight!(isImplicitlyConvertible, F), T))
+{
+    pragma(inline, true);
+
+    alias R = OriginalType!(Unqual!F);
+    static if (T.length <= 4)
+        return fimumImpl!(">", T.length, R)(nums);
+    else
+        return fimumImpl!(">", R)(nums);
+}
+
+/// ditto
+auto fmax(F)(const F[] nums) pure @safe nothrow @nogc
+    if (isFloatingPoint!F)
+{
+    pragma(inline, true);
+
+    alias R = OriginalType!(Unqual!F);
+    return fimumImpl!(">", R)(nums);
+}
+
+/// ditto
+auto fmax(F, size_t count)(auto ref const(F[count]) nums) pure @safe nothrow @nogc
+    if (isFloatingPoint!F)
+{
+    pragma(inline, true);
+
+    alias R = OriginalType!(Unqual!F);
+    static if (count <= 4)
+        return fimumImpl!(">", count, R)(nums);
+    else
+        return fimumImpl!(">", R)(nums[]);
+}
+
+///
+pure @safe nothrow @nogc unittest
+{
+    // Any number of arguments is supported:
+    assert(fmax(101.0, -5.0, 7.3723e121, double.epsilon) == 7.3723e121);
+    assert(fmax(1) == 1.0);
+    assert(fmax(ubyte.max, 25.0f) == 255.0f);
+    // Static and dynamic floating-point arrays, too:
+    double[7] nums = [0.0, double.min_normal, 0.5, -1.0, -double.max, double.infinity, 8];
+    assert(fmax(nums) == double.infinity);
+    assert(fmax(nums[1 .. 3]) == 0.5);
+
+    // NaN handling:
+    assert(fmax().isNaN);
+    assert(fmax(13, -431_315, long.max, real.nan, float.infinity).isNaN);
```

Comment:
One of the purposes of NaN values is to warn when a complex number is generated. This would be bad:
```D
assert(fmax(sqrt(-1.0), double.infinity) is double.infinity);
```
If we're going to have NaN poisoning, then we should have NaN poisoning (at least by default). Almost-but-not-quite poisoning will likely lead to more bug reports down the line.

If there is high demand for the above behaviour though, I suppose I could add another template parameter.